core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -471,7 +515,13 @@ static void setTokenRefreshNumRetries(int retries) {
      */
     public static AuthSession empty() {
       return new AuthSession(
-          ImmutableMap.of(), null, null, null, OAuth2Properties.CATALOG_SCOPE, null);
+          ImmutableMap.of(),
+          null,
+          null,
+          null,
+          OAuth2Properties.CATALOG_SCOPE,
+          null,
+          ImmutableMap.of());
```

Comment:
it has become quite painful to add new functionality/parameters to `AuthSession`, since it requires adding a single param at a bunch of places. The most recent change was done by #8976.

A while ago I experimented with an [`AuthConfig`](https://github.com/apache/iceberg/commit/3bed59907f4230d10a97742c58ab4b85ce7b1ebf#diff-d783470ff004c360395125d467f1aadd776c8c9e925ba52639a405a03a72b870R647), which would carry all parameters required for an `AuthSession`, so that it would be easier to add new configuration parameters to it. So I wonder if we should refactor `AuthSession` and its configuration so that it's less painful to add new stuff to it?

/cc @danielcweeks @amogh-jahagirdar @rdblue

Reply:
+1 to the painful to add parameters. With this PR, I have tried to add all optional oauth parameters (including scope) in a map but having a class like `[AuthConfig](https://github.com/apache/iceberg/commit/3bed59907f4230d10a97742c58ab4b85ce7b1ebf#diff-d783470ff004c360395125d467f1aadd776c8c9e925ba52639a405a03a72b870R647),` which holds all params with validation logic and passing it around would be best.

Reply:
Can we generate the request class(e.g., [OAuthClientCredentialsRequest](https://github.com/apache/iceberg/blob/bcbcbb263ea7e13ab22d0feb918e207c7e42dbbd/open-api/rest-catalog-open-api.yaml#L2919) ) from the spec so that we can pass a request object with all required and optional params? By doing that, we will always make sure the impl. synced with spec.